internal/mode/static/nginx/config/stream_servers_test.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/config/stream"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/dataplane"
+)
+
+func TestExecuteStreamServers(t *testing.T) {
+	conf := dataplane.Configuration{
+		TLSPassthroughServers: []dataplane.Layer4VirtualServer{
+			{
+				Hostname:     "example.com",
+				Port:         8081,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "example.com",
+				Port:         8080,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "cafe.example.com",
+				Port:         8080,
+				UpstreamName: "backend2",
+			},
+		},
+	}
+
+	expSubStrings := map[string]int{
+		"pass $dest8081;": 1,
+		"pass $dest8080;": 1,
+		"ssl_preread on;": 2,
+		"proxy_pass":      3,
+	}
+	g := NewWithT(t)
+
+	results := executeStreamServers(conf)
+	g.Expect(results).To(HaveLen(1))
+	result := results[0]
+
+	g.Expect(result.dest).To(Equal(streamConfigFile))
+	for expSubStr, expCount := range expSubStrings {
+		g.Expect(strings.Count(string(result.data), expSubStr)).To(Equal(expCount))
+	}
+}
+
+func TestCreateStreamServers(t *testing.T) {
+	conf := dataplane.Configuration{
+		TLSPassthroughServers: []dataplane.Layer4VirtualServer{
+			{
+				Hostname:     "example.com",
+				Port:         8081,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "example.com",
+				Port:         8080,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "cafe.example.com",
+				Port:         8080,
+				UpstreamName: "backend2",
+			},
+			{
+				Hostname:     "wrong.example.com",
+				Port:         8081,
+				UpstreamName: "",
+			},
```

Comment:
what if we have a single passthrough server for a port. will the logic still work?

for example:

```
			{
				Hostname:     "cafe.example.com",
				Port:         8083,
				UpstreamName: "backend2",
			},
```

Reply:
yep i think so. Why would it not work

Reply:
looks good 👍  I misunderstood the code.